src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -184,6 +184,15 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
                         genHWIntrinsicJumpTableFallback(intrinsicID, op3Reg, baseReg, offsReg, emitSwCase);
                     }
                 }
+                else if (category == HW_Category_MemoryStore)
+                {
+                    assert(intrinsicID == NI_SSE2_MaskMove);
+                    assert(targetReg == REG_NA);
+
+                    // SSE2 MaskMove hardcodes the destination (op3) in DI/EDI/RDI
+                    assert((op3Reg & RBM_EDI) != 0);
```

Comment:
Even if `op3Reg` was guaranteed to be in the correct register, this would not be correct, since `op3Reg` is a `regNumber` and `RBM_EDI` is a `regMaskTP`. In any case, you need to check to see if it is `REG_EDI` and if not generate a move.